Content.Server/Language/LanguageSystem.cs

```diff
@@ -0,0 +1,304 @@
+using System.Linq;
+using System.Text;
+using Content.Server.GameTicking.Events;
+using Content.Shared.Language;
+using Content.Shared.Language.Systems;
+using Robust.Shared.Random;
+using UniversalLanguageSpeakerComponent = Content.Shared.Language.Components.UniversalLanguageSpeakerComponent;
+
+namespace Content.Server.Language;
+
+public sealed partial class LanguageSystem : SharedLanguageSystem
+{
+    /// <summary>
+    ///   A random number added to each pseudo-random number's seed. Changes every round.
+    /// </summary>
+    public int RandomRoundSeed { get; private set; }
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LanguageSpeakerComponent, ComponentInit>(OnInitLanguageSpeaker);
+        SubscribeAllEvent<RoundStartingEvent>(it => RandomRoundSeed = _random.Next());
+
+        InitializeNet();
+    }
+
+    private void OnInitLanguageSpeaker(EntityUid uid, LanguageSpeakerComponent component, ComponentInit args)
+    {
+        if (string.IsNullOrEmpty(component.CurrentLanguage))
+        {
+            component.CurrentLanguage = component.SpokenLanguages.FirstOrDefault(UniversalPrototype);
+        }
+    }
+
+    /// <summary>
+    ///     Obfuscate speech of the given entity, or using the given language.
+    /// </summary>
+    /// <param name="source">The speaker whose message needs to be obfuscated. Must not be null if "language" is not set.</param>
+    /// <param name="language">The language for obfuscation. Must not be null if "source" is null.</param>
+    public string ObfuscateSpeech(EntityUid? source, string message, LanguagePrototype? language = null)
+    {
+        if (language == null)
+        {
+            if (source is not { Valid: true })
+            {
+                throw new NullReferenceException("Either source or language must be set.");
+            }
+            language = GetLanguage(source.Value);
+        }
+
+        var builder = new StringBuilder();
+        if (language.ObfuscateSyllables)
+        {
+            ObfuscateSyllables(builder, message, language);
+        }
+        else
+        {
+            ObfuscatePhrases(builder, message, language);
+        }
```

Comment:
I had a terrific idea. This should take a generic 'LanguageObfuscator' Class with children for each method of obfuscation. Instead of a binary boolean here, it would simply take the Obfuscator from the Language Prototype and execute the 'Obfuscate' function on it, using whatever is returned. This allows languages to implement custom obfuscation logic, for instance binary could be obfuscated based on *actually being converted to binary* (a stupid but funny suggestion), or Shadowkin could use whatever absurd system Death made.

The replacement portion of the YAML would be moved to the default obfuscator, since not every one would them.

Reply:
This is a good idea!
But that could be made in the future and not planned for the base code.